spring-integration-amqp/src/test/java/org/springframework/integration/amqp/outbound/RabbitStreamMessageHandlerTests.java

```diff
@@ -117,4 +125,108 @@ void sendNative() throws InterruptedException {
 		streamTemplate.close();
 	}
 
+	@Test
+	void errorChanelAsync() {
+		Environment env = Environment.builder()
+				.lazyInitialization(true)
+				.port(RabbitTestContainer.streamPort())
+				.build();
+		try {
+			env.deleteStream("stream.stream");
+		} catch (Exception e) {
+		}
+		env.streamCreator().stream("stream.stream").create();
+		RabbitStreamTemplate streamTemplate = new RabbitStreamTemplate(env, "stream.stream");
```

Comment:
I think for this test we don’t need to go to the broker to get failure back.
Consider to mock this template to throw something when we interact with it from the handler.